### PR TITLE
KEYCLOAK-12316 Simplify conditions in packages: authentication, broker, credential, protocol from module "services"

### PR DESCRIPTION
**Pull request by Captain1653**

### CODE REVIEW

**Review by martin-kanis — APPROVED** [review]
@Captain1653 LGTM thank you

File: services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProviderConfig.java [hunk 1]
```diff
@@ -103,7 +103,7 @@ public void setBackchannelSupported(boolean backchannel) {
 
     public boolean isDisableUserInfoService() {
         String disableUserInfo = getConfig().get("disableUserInfo");
-        return disableUserInfo == null ? false : Boolean.valueOf(disableUserInfo);
+        return Boolean.parseBoolean(disableUserInfo);
```
Comment (stianst) [inline services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProviderConfig.java]: This doesn't count for disableUserInfo could be null

Comment (Captain1653) [reply services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProviderConfig.java]: Maybe I'm wrong, but javadoc Boolean.parseBoolean says:

> Parses the string argument as a boolean.  The {@code boolean}
     * returned represents the value {@code true} if the string argument
     * is not {@code null} and is equal, ignoring case, to the string
     * {@code "true"}. <p>

The code Boolean.parseBoolean is:
`return ((s != null) && s.equalsIgnoreCase("true"));`

Comment (stianst) [reply services/src/main/java/org/keycloak/broker/oidc/OIDCIdentityProviderConfig.java]: You're correct. I read the JavaDoc now, but to be honest it's not fully clear what the behaviour is with null, so I checked and it does indeed return false.

**Review by stianst — APPROVED** [review]

**Pull request merged**